### PR TITLE
Add localize command handle

### PR DESCRIPTION
**Pull request by annasong20**
This PR implements the handle that exposes `localize` as a command to users. 

In order to test the command handle code, this PR moves existing test helpers in `krusty` to `api/testutils`.

### CODE REVIEW

**Comment by annasong20** [comment]
@natasha41575 Does the failed presubmit test mean that we need to release a new version of `api` before I can merge this PR?

File: kustomize/commands/localize/localize.go
Comment (natasha41575) [inline kustomize/commands/localize/localize.go:54]: We should probably mark this as alpha the same way we do the other alpha commands. For example:

```
$ kustomize cfg -h
Commands for reading and writing configuration

Usage:
  kustomize cfg [command]

Available Commands:
  cat           [Alpha] Print Resource Config from a local directory.
  count         [Alpha] Count Resources Config from a local directory.
  grep          [Alpha] Search for matching Resources in a directory or from stdin
  tree          [Alpha] Display Resource structure from a directory or stdin.
```

File: kustomize/commands/localize/localize.go
Comment (natasha41575) [inline kustomize/commands/localize/localize.go:68]: Is it standard practice in k/k to spit out a warning for alpha commands every time they are run? The current ones we have (e.g. `kustomize cfg cat`) don't, and it doesn't seem necessary to me. @KnVerey do you have an opinion on this?

Comment (natasha41575) [inline kustomize/commands/localize/localize.go:68]: May be a moot point with my other comment, but warnings should be logged to `stderr`, not `stdout`.

File: kustomize/commands/localize/flagscope.go
Comment (natasha41575) [inline kustomize/commands/localize/flagscope.go:14]: Creating a whole separate file/function for a single flag seems unnecessary to me, It might make more sense to do this inline.

File: kustomize/commands/localize/localize.go
Comment (natasha41575) [inline kustomize/commands/localize/localize.go:57]: With all our other commands, we write out the full command in the example (so in this case, `kustomize localize`), instead of using %s placeholders. It is a little cleaner and easier to follow. It is very unlikely that these names will ever change, and if they do, updating the examples will be the least of our worries.

**Review by natasha41575 — COMMENTED** [review]
/hold

We should make sure we have everything we want to add to alpha `kustomize localize` before we expose it in the CLI.

File: api/testutils/loctest/helper.go
Comment (yuwenma) [reply api/testutils/loctest/helper.go:4]: +1. I like the suggestion. "helper" and "util" are always less preferred due to its vague meaning.

**Comment by yuwenma** [comment]
@annasong20 I think you may have to split this PR into two parts, having the `./api` PR merges first and then `./kustomize` PR considering that these two libraries are released separately.

**Comment by annasong20** [comment]
> @annasong20 I think you may have to split this PR into two parts, having the `./api` PR merges first and then `./kustomize` PR. Because these two libraries are released separately, if regression happens, one will affect the other.

@yuwenma excellent suggestion! #4974 now handles the `api` changes: moving the test helpers from `api/krusty/localizer` to `api/testutils/localizertest`.

File: api/testutils/loctest/helper.go
Comment (annasong20) [reply api/testutils/loctest/helper.go:4]: I moved the `api` test helper changes to #4974: https://github.com/kubernetes-sigs/kustomize/pull/4959#issuecomment-1380826106

@natasha41575 renamed package!
@yuwenma I renamed the file to `filesystem.go` because all the test helpers deal with the real file system. Let me know if you have a better name in mind!

**Comment by annasong20** [comment]
@natasha41575, @yuwenma, @KnVerey I cherry-picked the relevant commits from #4974, #4975 to demonstrate what the final code would look like. 

However, we should look at those 2 PRs first. After/If those PRs merge, I will rebase this one off master.

File: kustomize/commands/localize/localize.go
Comment (annasong20) [reply kustomize/commands/localize/localize.go]: Done. I added `[Alpha]` to both the short and long descriptions of the command.

File: kustomize/commands/localize/localize.go
Comment (annasong20) [reply kustomize/commands/localize/localize.go]: I removed the warning for the `[Alpha]` prefix in the command descriptions.

Comment (annasong20) [reply kustomize/commands/localize/localize.go]: Warning message removed.

File: kustomize/commands/localize/flagscope.go
Comment (annasong20) [reply kustomize/commands/localize/flagscope.go]: Done.

File: kustomize/commands/localize/localize.go
Comment (annasong20) [reply kustomize/commands/localize/localize.go]: Done. I replaced the placeholders with the written string.

File: kustomize/commands/localize/localize.go
Comment (annasong20) [reply kustomize/commands/localize/localize.go]: Removed.

**Review by natasha41575 — APPROVED** [review]
I built off this branch and played around with `kustomize localize` and it's looking good. I think one thing could be improved about the UX is I would like to see some sort of success message after running `kustomize localize`. In particular, if I don't provide the destination argument, it would serve as an easy place for the user to see the name of the localized directory. Even if I do provide the destination argument, it's good feedback for the user to see that everything went as expected.

e.g. it would be nice to see
```
$ kustomize localize remote
Successfully localized `remote` into directory `localized-remote`.
```

Other than this small comment, this PR LGTM. I know we are waiting on https://github.com/kubernetes-sigs/kustomize/pull/4975, so I can remove the hold on this when that goes in.

**Comment by annasong20** [comment]
> I built off this branch and played around with `kustomize localize` and it's looking good. I think one thing could be improved about the UX is I would like to see some sort of success message after running `kustomize localize`. In particular, if I don't provide the destination argument, it would serve as an easy place for the user to see the name of the localized directory. Even if I do provide the destination argument, it's good feedback for the user to see that everything went as expected.
> 
> e.g. it would be nice to see
> 
> ```
> $ kustomize localize remote
> Successfully localized `remote` into directory `localized-remote`.
> ```
> 
> Other than this small comment, this PR LGTM. I know we are waiting on #4975, so I can remove the hold on this when that goes in.

Thanks, @natasha41575! I exposed the destination directory path in #5010. Once that merges, I'll rebase and use the returned destination in the `Success` message.

**Pull request merged**
